test/e2e/app-dir/app-middleware/middleware.js

```diff
@@ -44,6 +44,21 @@ export async function middleware(request) {
     return NextResponse.rewrite(request.nextUrl)
   }
 
+  if (request.nextUrl.pathname === '/rsc-cookies') {
+    const res = NextResponse.next()
+    res.cookies.set('rsc-cookie-value-1', `${Math.random()}`)
+    res.cookies.set('rsc-cookie-value-2', `${Math.random()}`)
```

Comment:
Should this also work with `cookies()` directly (not from `res`)?

Reply:
Yeah perhaps -- although writing via `cookies()` has been documented as only working in server actions/route handlers. So I think making this change could be separate from this while making the canonical middleware use-case work correctly.